Mathlib/Tactic/SolveByElim.lean

```diff
@@ -99,6 +73,114 @@ structure Config extends ApplyConfig where
   If failure, we backtrack. (defaults to failure) -/
   discharge : MVarId → MetaM (Option (List MVarId)) := fun _ => failure
 
+
+def backtracking (cfg : Config) (alternatives : MVarId → MetaM (List (MetaM (List MVarId))))
```

Comment:
Could `backtracking` be given a `trace` argument like `applyTactics` for use as a component in other tactics? (Or is it meant just for `solve_by_elim`? If so, should it be in the `SolveByElim` namespace?)